src/software/jetson_nano/thunderloop.cpp

```diff
@@ -124,7 +135,14 @@ Thunderloop::~Thunderloop() {}
 
                 // Update primitive executor's primitive set
                 {
+                    struct timespec time_since_last_primitive_received;
+                    clock_gettime(CLOCK_MONOTONIC, &current_time);
+                    ScopedTimespecTimer::timespecDiff(
+                        &current_time, &last_primitive_received_time,
+                        &time_since_last_primitive_received);
                     clock_gettime(CLOCK_MONOTONIC, &last_primitive_received_time);
+                    network_status_.set_ms_since_last_primitive_received(
+                        time_since_last_primitive_received.tv_nsec / 1000000);
```

Comment:
nit: Should use existing constants for time conversions (same with conversions below)
```suggestion
                        time_since_last_primitive_received.tv_nsec * static_cast<int>(MILLISECONDS_PER_NANOSECOND));
```